src/Campaigns/Actions/RedirectDisabledCampaignPage.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+namespace Give\Campaigns\Actions;
+
+use Give\Campaigns\Models\CampaignPage;
+
+/**
+ * @unreleased
+ */
+class RedirectDisabledCampaignPage
+{
+    /**
+     * @unreleased
+     */
+    public function __invoke()
+    {
+        if('give_campaign_page' !== get_post_type()) {
+            return;
+        }
+
+        $campaignPage = CampaignPage::find(get_the_ID());
+
+        if(!$campaignPage->campaign()->enableCampaignPage) {
```

Comment:
To prevent a fatal error if `$campaign` is not found:

```suggestion
        if($campaignPage && !$campaignPage->campaign()->enableCampaignPage) {
```